### PR TITLE
Prevent selecting service account in the extension.

### LINKED ISSUE #5: Consider Service Account
## Context

The extension currently lists the service account as one of the accounts.
Users have been accidentally deploying code to the account, breaking the emulator.
We should probably not allow deploying to the service account or remove it from the account list.

## Definition of Done

- TBD

### PR DESCRIPTION
**Pull request by 10thfloor**
Closes #5 

## Description

- Hides service account address from the UI
- Selects the first default account, once accounts have been bootstrapped after emulator start

### CODE REVIEW

**Comment by 10thfloor** [comment]
@turbolent Is it still useful to display the service account address somewhere in the extension? Users can see the address in the emulator output if necessary.

File: src/commands.ts [hunk 1]
```diff
@@ -13,148 +21,166 @@ export const SWITCH_ACCOUNT = "cadence.switchActiveAccount";
...
-            window.showInformationMessage(`Switched to account ${activeAccount.fullName()}`);
-
-            renderExtension(ext);
+  // Suffix to indicate which account is active
+  const activeSuffix = "(active)";
+  // Create the options (mark the active account with an 'active' prefix)
+  console.log(">>>>", ext.config.accounts);
+  const accountOptions = Object.values(ext.config.accounts)
+    // Mark the active account with a `*` in the dialog
+    .filter((acct) => acct.address !== SERVICE_ADDR)
```
Comment (10thfloor) [inline src/commands.ts:126]: This is the line you care about.

File: src/commands.ts [hunk 2]
```diff
@@ -13,148 +21,166 @@ export const SWITCH_ACCOUNT = "cadence.switchActiveAccount";
...
+  // create default accounts after the emulator has started
+  // skip service account since it is already created
+  setTimeout(async () => {
+    try {
+      const accounts = await ext.api.createDefaultAccounts(
+        ext.config.numAccounts - 1
+      );
+      accounts.forEach((address) => ext.config.addAccount(address));
+      // set the first account as active
+      ext.config.setActiveAccount(1);
```
Comment (10thfloor) [inline src/commands.ts:84]: ^^ This is the line you care about.

File: src/commands.ts [hunk 3]
```diff
@@ -13,148 +21,166 @@ export const SWITCH_ACCOUNT = "cadence.switchActiveAccount";
...
-
-            ext.config.setActiveAccount(activeIndex)
-
-            window.showInformationMessage(`Switched to account ${activeAccount.fullName()}`);
-
-            renderExtension(ext);
+  // Suffix to indicate which account is active
+  const activeSuffix = "(active)";
+  // Create the options (mark the active account with an 'active' prefix)
+  console.log(">>>>", ext.config.accounts);
```
Comment (turbolent) [inline src/commands.ts:123]:
```suggestion
```

File: src/commands.ts [hunk 4]
```diff
@@ -13,148 +21,166 @@ export const SWITCH_ACCOUNT = "cadence.switchActiveAccount";
...
+  const accountOptions = Object.values(ext.config.accounts)
+    // Mark the active account with a `*` in the dialog
+    .filter((acct) => acct.address !== SERVICE_ADDR)
+    .map((account) => {
+      const suffix: String =
+        account.index === ext.config.activeAccount ? ` ${activeSuffix}` : "";
+      const label = `${account.fullName()}${suffix}`;
```
Comment (turbolent) [inline src/commands.ts:130]:
```suggestion
      const label = account.fullName() + suffix;
```

File: .vscode/launch.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (turbolent) [inline .vscode/launch.json:1]: 👍

Comment (10thfloor) [reply .vscode/launch.json:1]: Updated!

File: src/commands.ts [hunk 6]
```diff
@@ -13,148 +21,165 @@ export const SWITCH_ACCOUNT = "cadence.switchActiveAccount";
...
+  ext.terminal.show();
+
+  // create default accounts after the emulator has started
+  // skip service account since it is already created
+  setTimeout(async () => {
+    try {
+      const accounts = await ext.api.createDefaultAccounts(
+        ext.config.numAccounts - 1
+      );
+      accounts.forEach((address) => ext.config.addAccount(address));
```
Comment (psiemens) [inline src/commands.ts:82]: Can we just filter the service account here?

Comment (10thfloor) [reply src/commands.ts:82]: I think by this point the service account is already created. This list only holds the 2 "default accounts". 
I'll double check.

Comment (psiemens) [reply src/commands.ts:82]: Ah right, my bad.

**Pull request merged**